component/local/file/file.go

```diff
@@ -136,6 +147,7 @@ func (c *Component) Run(ctx context.Context) error {
 	for {
 		select {
 		case <-ctx.Done():
+			c.opts.Registerer.Unregister(c.lastAccessed)
```

Comment:
This isn't guaranteed to be called if the component never starts up (ie because construction failed)